### PR TITLE
Javascript tests

### PR DESCRIPTION
**Pull request by prescod**

### CODE REVIEW

File: src/js/store/perfdata/actions.js [hunk 1]
```diff
@@ -38,39 +47,52 @@ export const perfRESTFetch = (url: string, params?: {}): ThunkAction => (
...
+  apiFetch,
+  prefix,
+  url,
+  checkValid,
+}) => {
+  dispatch({ type: `${prefix}_DATA_LOADING`, payload: { url } });
+  return apiFetch(url, { method: 'GET' }).then(payload => {
+    try {
+      if (!payload){ throw new Error('No payload');} // prettier-ignore
+      if (payload.error){ throw new Error(`Server error: ${payload.error}`);} // prettier-ignore
```
Comment (davisagli) [inline src/js/store/perfdata/actions.js]: I think it'd be better to follow the format prettier wants here. I find single-line conditionals less readable even without the extra comment.

File: src/js/store/perfdata/actions.js [hunk 2]
```diff
@@ -38,39 +47,52 @@ export const perfRESTFetch = (url: string, params?: {}): ThunkAction => (
...
+  perfREST_API({
+    dispatch,
+    _getState,
+    apiFetch,
+    prefix: 'UI',
+    url: testmethod_perf_UI_url,
+    checkValid: assertUIData,
+  });
+
+// $FlowFixMe
```
Comment (davisagli) [inline src/js/store/perfdata/actions.js]: 😜

File: src/test/js/api/testmethod_perf_UI.json [hunk 3]
```diff
@@ -0,0 +1,1752 @@
+{
+  "buildflow_filters": [
+     {
+        "name": "repo",
+        "label": "Repo Name",
+        "field_type": "ChoiceField",
+        "field_module": "django_filters.fields",
+        "lookup_expr": "exact",
+        "choices": [
```
Comment (davisagli) [inline src/test/js/api/testmethod_perf_UI.json]: Please update this to remove private repositories.

**Comment by davisagli** [comment]
I guess the warning I mentioned above is the expected behavior if there's an unknown field type.

I'd still like to get the tests running in Heroku CI and reporting to coveralls now rather than later unless it is challenging. Let me know if you want help with this.

**Comment by prescod** [comment]
I'll open a new ticket for myself to automate the Heroku aspects. Just in case it has challenges I'd rather not have this PR hanging open. @davisagli

**Pull request merged**